server/shared/src/main/scala/org/http4s/server/middleware/GZip.scala

```diff
@@ -20,18 +20,17 @@ package middleware
 
 import cats.Functor
 import cats.data.Kleisli
-import cats.effect.Sync
 import cats.syntax.all._
 import fs2.compression._
 import org.http4s.headers._
 import org.log4s.getLogger
 
-object GZip {
+object GZip extends GZipPlatform {
   private[this] val logger = getLogger
 
   // TODO: It could be possible to look for F.pure type bodies, and change the Content-Length header after
   // TODO      zipping and buffering all the input. Just a thought.
-  def apply[F[_]: Functor, G[_]: Sync](
+  def apply[F[_]: Functor, G[_]: Compression](
```

Comment:
This is source-compatible for JVM, because anyone who already had `Sync` automatically has a `Compression` in scope too. Binary-compatibility is preserved via a deprecated overload in `GZipPlatform`.